File: Q16.java
import java.util.Scanner;

public class Q16
{
    public static Scanner leia = new Scanner(System.in);
    public static void main(String args[])
    {
        float número[] = new float[20];
        int vezes = 1, quantidade[] = {0, 0, 0};
        for (int i = 0; i < número.length; i ++, vezes ++)
        {
            System.out.printf("Digite o %d° número: ", vezes);
            número[i] = leia.nextFloat();
            if (número[i] >= 0 & número[i] <= 100)
            {
                quantidade[0] ++;
            }
            else if (número[i] >= 101 & número[i] <= 200)
            {
                quantidade[1] ++;
            }
            else if (número[i] > 200)
            {
                quantidade[2] ++;
            }
        }
        if (quantidade[0] + quantidade[1] + quantidade[2] == 19)
        {
            System.out.printf("Entre 0 e 100: %d, entre 101 e 200: %d e maiores do que 200: %d.\n", quantidade[0], quantidade[1], quantidade[2]);
            System.out.println("O número restante é menor do que 0.");
        }
        else if (quantidade[0] + quantidade[1] + quantidade[2] < 19)
        {
            System.out.printf("Entre 0 e 100: %d, entre 101 e 200: %d e maiores do que 200: %d.\n", quantidade[0], quantidade[1], quantidade[2]);
            System.out.println("Os números restantes são menores do que 0.");
        }
        else
        {
            System.out.printf("Entre 0 e 100: %d, entre 101 e 200: %d e maiores do que 200: %d.\n", quantidade[0], quantidade[1], quantidade[2]);
        }
    }
}

File: Q16.py
número = list(range(20))
vezes = 1
quantidade = [0, 0, 0]
for i in range(0, len(número)):
    número[i] = float(input(f"Digite o {vezes}° número: "))
    if número[i] >= 0 and número[i] <= 100:
        quantidade[0] += 1
    elif número[i] >= 101 and número[i] <= 200:
        quantidade[1] += 1
    elif número[i] > 200:
        quantidade[2] += 1
    vezes += 1
if quantidade[0] + quantidade[1] + quantidade[2] == 19:
    print(f"Entre 0 e 100: {quantidade[0]}, entre 101 e 200: {quantidade[1]} e maiores do que 200: {quantidade[2]}.")
    print("O número restante é menor do que 0.")
elif quantidade[0] + quantidade[1] + quantidade[2] < 19:
    print(f"Entre 0 e 100: {quantidade[0]}, entre 101 e 200: {quantidade[1]} e maiores do que 200: {quantidade[2]}.")
    print("Os números restantes são menores do que 0.")
else:
    print(f"Entre 0 e 100: {quantidade[0]}, entre 101 e 200: {quantidade[1]} e maiores do que 200: {quantidade[2]}.")